partner_compassion/model/partner_compassion.py

```diff
@@ -8,79 +8,76 @@
 #    The licence is in the file __openerp__.py
 #
 ##############################################################################
-from openerp.osv import fields, orm
-from openerp.tools.translate import _
+
+from openerp import api, fields, models, _
 
 # fields that are synced if 'use_parent_address' is checked
 ADDRESS_FIELDS = [
     'street', 'street2', 'street3', 'zip', 'city', 'state_id', 'country_id']
 
 
-class ResPartner(orm.Model):
+class ResPartner(models.Model):
     """ This class upgrade the partners to match Compassion needs.
     """
 
     _inherit = 'res.partner'
 
-    def _lang_get(self, cr, uid, context=None):
-        lang_obj = self.pool.get('res.lang')
-        ids = lang_obj.search(cr, uid, [], context=context)
-        res = lang_obj.read(cr, uid, ids, ['code', 'name'], context)
-        return [(r['code'], r['name']) for r in res]
+    ##########################################################################
+    #                        NEW PARTNER FIELDS                              #
+    ##########################################################################
+
+    ref = fields.Char(default=lambda self: self.env[
+        'ir.sequence'].get('partner.ref'))
+    street3 = fields.Char("Street3", size=128)
+    member_ids = fields.One2many(
+        'res.partner', 'church_id', 'Members',
+        domain=[('active', '=', True)])
+    is_church = fields.Boolean(
+        string="Is a Church", compute='_is_church', store=True)
+    church_id = fields.Many2one(
+        'res.partner', 'Church', domain=[('is_church', '=', True)])
+    church_unlinked = fields.Char(
+        "Church (N/A)",
+        help=_("Use this field if the church of the partner"
+               " can not correctly be determined and linked."))
+    deathdate = fields.Date('Death date')
+    birthdate = fields.Date('Birthdate')
+    lang = fields.Selection(
+        string='Language', compute='_lang_get', required=True,
```

Comment:
lang is not a computed field, it is a function for returning the selection values.
so '_lang_get' should be the first parameter (not compute parameter).